sw/device/silicon_creator/lib/epmp.c

```diff
@@ -56,5 +57,10 @@ rom_error_t epmp_state_check(const epmp_state_t *s) {
 
 #undef CHECK_CSR
 
-  return result ? kErrorOk : kErrorEpmpBadCheck;
+  const uint32_t total = 22;
+  rom_error_t error = kErrorEpmpBadCheck;
+  if (launder32(checks) == total) {
+    error = (kErrorOk ^ total) ^ checks;
+  }
+  return error;
```

Comment:
Should we add some redundant checks? For example:

```suggestion
  enum { kTotalChecks = 22 };
  // Hamming distance of 3, error = 0x72f, kErrorOk = 0x739.
  rom_error_t error = kErrorOk ^ kTotalChecks;
  if (launder32(checks) == kTotalChecks) {
    HARDENED_CHECK_EQ(checks, kTotalChecks);
    error ^= checks;
    HARDENED_CHECK_EQ(error, kErrorOk);
    return error;
  }
  return kErrorEpmpBadCheck;
```

I guess we could initialize `error` to some other value and xor the difference from kErrorOk as well but this already has a HD of 3 from `kErrorOk`.

Reply:
Done, thanks @alphan!